nullaway/src/test/resources/com/uber/nullaway/testdata/CheckContractNegativeCases.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2017 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.testdata;
+
+import javax.annotation.Nullable;
+import org.jetbrains.annotations.Contract;
+
+public class CheckContractNegativeCases {
+
+  @Contract("_, !null -> !null")
+  @Nullable
+  Object foo(Object a, @Nullable Object b) {
+    if (a != null) {
+      return b;
+    }
+    return new Object();
```

Comment:
Currently, we are not using the information from the method signature. I don't know why I didn't do it right now. Let me try if this can be done easily..

Reply:
For the first one: I think checking the information from the method signature is important. I don't see us turning this on internally without that, based on the methods I am seeing that use `@Contract` in our codebase :) 

The second is perfectly reasonable given the size of this PR already

Reply:
IIRC we can / should only due heuristic checking of contracts like `null -> false`, as that requires tracking values of boolean variables / expressions.  We could at least catch cases where the contract is `null -> false` but the actual behavior is `null -> true`, where the return is literally `return true`.  I agree this can be added in a follow-up

Reply:
We can't check 100%, but we can easily do at least everything the stream handler does for `.filter(...)` and boolean returns. It won't cover all cases, but many of this methods are just:

```
@Contract("null -> false")
boolean foo(Object o) {
   // Some code
  if (o == null) {
    // Some error handling
    return false;
  }
  // Some complex logic
}
```

And in that case, we can fully guarantee `null -> false` ;)

Reply:
The first part is Done 👍🏻